### ISSUE TITLE
Can't build project on linux after changing SDK 2.0 -> 2.1

### ISSUE THREAD

**Issue by kolipka** [dotnet/cli#10088]
## Steps to reproduce
I had dotnet core 2.0 installed on Ubuntu 17. Removed it and installed 2.1 SDK. When I try to build 
netcoreapp2.1 app it fails. It builds on Windows.

When I run `dotnet new webapi` I get old 2.0 template. On windows 2.1.

## Expected  behavior
project builds

## Actual behavior
         /usr/share/dotnet/sdk/2.1.4/Sdks/Microsoft.NET.Sdk/build/Microsoft.NET.TargetFrameworkInference.targets(135,5): error : The current .NET SDK does not support targeting .NET Core 2.1.  Either target .NET Core 2.0 or lower, or use a version of the .NET SDK that supports .NET Core 2.1. [/home/konrad/Projects/Logix.API/Logix.API/Logix.API.csproj]

## Environment data
`dotnet --info` output:
```
.NET Command Line Tools (2.1.4)

Product Information:
 Version:            2.1.4
 Commit SHA-1 hash:  5e8add2190

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  17.10
 OS Platform: Linux
 RID:         ubuntu.17.10-x64
 Base Path:   /usr/share/dotnet/sdk/2.1.4/

Host (useful for support):
  Version: 2.1.4

  Commit:  85255dde3e

.NET Core SDKs installed:
  2.1.4 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.4 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.4 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.4 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by dasMulli** [comment]
You will need to install a newer SDK version. only SDKs beginning at 2.1.300 can build .NET Core 2.1 projects. 2.1.4 is still a .NET Core 2.0 version (with confusing naming)

**Comment by kolipka** [comment]
ooook. So one more thing: apt-get does not have new version yet? I can download .NET Core 2.1 SDK - Linux x64 Binaries (v2.1.403) from web page, apt-get feed seems to be outdated:

```bash
konrad@DEV-SERVER:~$ sudo apt-get update
Hit:1 http://pl.archive.ubuntu.com/ubuntu artful InRelease
Hit:2 http://pl.archive.ubuntu.com/ubuntu artful-updates InRelease
Hit:3 http://pl.archive.ubuntu.com/ubuntu artful-backports InRelease
Hit:4 http://security.ubuntu.com/ubuntu artful-security InRelease
Hit:5 https://deb.nodesource.com/node_9.x artful InRelease
Hit:6 https://packages.microsoft.com/repos/microsoft-ubuntu-artful-prod artful InRelease
Get:7 https://packages.microsoft.com/ubuntu/17.10/prod artful InRelease [2846 B]
Get:8 https://packages.microsoft.com/ubuntu/17.10/prod artful/main amd64 Packages [28,7 kB]
Fetched 31,6 kB in 0s (39,4 kB/s)
Reading package lists... Done
konrad@DEV-SERVER:~$ apt search dotnet-sdk
Sorting... Done
Full Text Search... Done
dotnet-sdk-2.0.2/artful,artful 2.0.2-1 amd64
  Microsoft .NET Core SDK - 2.0.2

dotnet-sdk-2.0.3/artful,artful 2.0.3-1 amd64
  Microsoft .NET Core SDK - 2.0.3

dotnet-sdk-2.1/artful,artful 2.1.302-1 amd64
  Microsoft .NET Core SDK 2.1.302

dotnet-sdk-2.1.100/artful,artful 2.1.100-1 amd64
  Microsoft .NET Core SDK - 2.1.100

dotnet-sdk-2.1.101/artful,artful 2.1.101-1 amd64
  Microsoft .NET Core SDK - 2.1.101

dotnet-sdk-2.1.102/artful,artful 2.1.102-1 amd64
  Microsoft .NET Core SDK - 2.1.102

dotnet-sdk-2.1.103/artful,artful 2.1.103-1 amd64
  Microsoft .NET Core SDK - 2.1.103

dotnet-sdk-2.1.104/artful,artful 2.1.104-1 amd64
  Microsoft .NET Core SDK - 2.1.104

dotnet-sdk-2.1.105/artful,artful 2.1.105-1 amd64
  Microsoft .NET Core SDK - 2.1.105

dotnet-sdk-2.1.2/artful,artful 2.1.2-1 amd64
  Microsoft .NET Core SDK - 2.1.2

dotnet-sdk-2.1.200/artful,artful 2.1.200-1 amd64
  Microsoft .NET Core SDK - 2.1.200

dotnet-sdk-2.1.201/artful,artful 2.1.201-1 amd64
  Microsoft .NET Core SDK - 2.1.201

dotnet-sdk-2.1.202/artful,artful 2.1.202-1 amd64
  Microsoft .NET Core SDK - 2.1.202

dotnet-sdk-2.1.3/artful,artful 2.1.3-1 amd64
  Microsoft .NET Core SDK - 2.1.3

dotnet-sdk-2.1.300-preview1-008174/artful,artful 2.1.300-preview1-008174-1 amd64
  Microsoft .NET Core SDK 2.1.300 - Preview

dotnet-sdk-2.1.300-preview2-008533/artful,artful 2.1.300-preview2-008533-1 amd64
  Microsoft .NET Core SDK 2.1.300 - Preview

dotnet-sdk-2.1.300-rc1-008673/artful,artful 2.1.300-rc1-008673-1 amd64
  Microsoft .NET Core SDK 2.1.300 - rc1

dotnet-sdk-2.1.4/artful,artful,now 2.1.4-1 amd64 [installed]
  Microsoft .NET Core SDK - 2.1.4


```

**Comment by livarcocc** [comment]
@leecow to comment on the state of the feed.

**Comment by leecow** [comment]
17.10 went end-of-life in July so the last available 2.1 SDK version available in the feed is 2.1.302. You could try installing 2.1.403 using the tar.gz but that's untested and unsupported because the OS is out of support. 

See the [os lifecycle policy](https://github.com/dotnet/core/blob/master/os-lifecycle-policy.md) for details.

**Comment by livarcocc** [comment]
Thanks @leecow
